packages/feedback2/src/core/integration.ts

```diff
@@ -0,0 +1,285 @@
+      throw new Error('Unable to attach to target element');
+    }
+
+    let dialog: Dialog | null = null;
+    const handleClick = async (): Promise<void> => {
+      if (!dialog) {
+        dialog = await _loadAndRenderDialog({
+          ...mergedOptions,
+          onFormClose: () => {
+            dialog && dialog.close();
+            mergedOptions.onFormClose && mergedOptions.onFormClose();
+          },
+          onFormSubmitted: () => {
+            dialog && dialog.removeFromDom();
+            mergedOptions.onFormSubmitted && mergedOptions.onFormSubmitted();
+          },
+        });
+      }
+      dialog.appendToDom();
+      dialog.open();
+    };
+    targetEl.addEventListener('click', handleClick);
+    const unsubscribe = (): void => {
+      _subscriptions = _subscriptions.filter(sub => sub !== unsubscribe);
+      dialog && dialog.removeFromDom();
+      dialog = null;
+      targetEl.removeEventListener('click', handleClick);
+    };
+    _subscriptions.push(unsubscribe);
+    return unsubscribe;
+  };
+
+  const autoInjectActor = (): void => {
+    const shadow = _createShadow(_options);
+    const actor = Actor({ buttonLabel: _options.buttonLabel, shadow });
+    const mergedOptions = mergeOptions(_options, {
+      onFormOpen() {
+        actor.removeFromDom();
+      },
+      onFormClose() {
+        actor.appendToDom();
+      },
+      onFormSubmitted() {
+        actor.appendToDom();
+      },
+    });
+    attachTo(actor.el, mergedOptions);
+
+    actor.appendToDom();
+  };
+
+  return {
+    name: 'Feedback2',
+    setupOnce() {
+      if (!isBrowser() || !_options.autoInject) {
+        return;
+      }
+
+      autoInjectActor();
+    },
+
+    /**
+     * Adds click listener to the element to open a feedback dialog
+     *
+     * The returned function can be used to remove the click listener
+     */
+    attachTo,
+
+    /**
+     * Creates a new widget. Accepts partial options to override any options passed to constructor.
+     */
+    async createWidget(optionOverrides: OverrideFeedbackConfiguration = {}): Promise<Dialog> {
+      return _loadAndRenderDialog(mergeOptions(_options, optionOverrides));
+    },
+
+    /**
+     * Removes the Feedback integration (including host, shadow DOM, and all widgets)
+     */
+    remove(): void {
+      if (_shadow) {
+        _shadow.parentElement && _shadow.parentElement.remove();
+        _shadow = null;
+      }
+      // Remove any lingering subscriptions
+      _subscriptions.forEach(sub => sub());
+      _subscriptions = [];
+    },
+  };
+}) satisfies IntegrationFn;
+
+export const feedback2Integration = defineIntegration(_feedback2Integration);
+
+/**
+ * @deprecated Use `feedback2ScreenshotIntegration()` instead
+ */
+// eslint-disable-next-line deprecation/deprecation
+export const Feedback2 = convertIntegrationFnToClass('Feedback2', feedback2Integration) as IntegrationClass<
```

Comment:
No need for this anymore, you can remove the class export here :)

Reply:
Consider it done!